juju/juju
# Improvements to caas operator password/storage/config

## Description of change

When deploying a k8s charm, it is now mandatory that a storage pool for operator storage be configured - operators require storage for charm state etc. This change also allows us to provide a writable agent config file. The file is provided read-only in a config map and copied to the writable filesystem. This then allows password and logging config to be properly managed.

Part of the change is that the address watcher plugin is removed from the caas operator provisioner worker and facades and moved across to the caas operator itself (and associated facades). Also, the api addresses are returned as part of the caas operator provisioning info call.

The deploy command has checks added to ensure that the required storage setup is all in place.

## QA steps

deploy a k8s charm with storage
ensure the necessary checks are done and errors if no storage pools are configured
ensure that the charm deploys ok when all necessary storage config is in place

## Documentation changes

discourse post will be updated
